### ISSUE TITLE
Add option for generic Write-Progress for long running commands/tools

### ISSUE THREAD

**Issue by DennisL68** [PowerShell/Crescendo#193]
### Summary of the new feature / enhancement

As a user of a familiar command that takes long time to finish
I want some kind of feedback during execution
So that I know the command is (probably) still processing

### Proposed technical implementation details (optional)

Invoke a Write-Progress feature just before starting the command if a switch for 'ShowProgress' is set in a command parameter section.
End the Write-Progress before the command moves over to the parse handler.

~~~ Mock up
$ProgressBlock = {
[console]::WindowHeight = 2
#[console]::BufferHeight = 2

[console]::WindowWidth = 40
[console]::BufferWidth = 40

[console]::Title = '<Command>'

while ($true) {
  1..100 | foreach {
    Write-Progress -Activity " Processing" -Status "`0" -PercentComplete $_
    Start-Sleep -Milliseconds 50
  }
}
}

$Progress = Start-Process PWSH "-noprofile -command $ProgressBlock" -PassThru

<command with args>

$Progress.Kill()
~~~

**Comment by theJasonHelmick** [comment]
@DennisL68 - Thank you for the suggestion.  This can be accomplished today by editing the generated module and adding your own additional progress.  Crescendo is a framework to accelerate the wrapping of native commands and allow you to use the full features of PowerShell in the generated module. Crescendo itself does not provide json configuration for every feature - as it's intended to get you started.

**Comment by DennisL68** [comment]
Yes, but I don't think this is a one-of case. 
Surely, there are many native commands that might take "for ever" to run that would benefit of such a feature built into Crescendo?

**Comment by theJasonHelmick** [comment]
@DennisL68 - Thank you. Because this is a feature you can already enable using PowerShell, this is not a priority for Crescendo at this time.

**Comment by theJasonHelmick** [comment]
@DennisL68 - Thank you. Because this is a feature you can already enable using PowerShell, this is not a priority for Crescendo at this time.
